Bugzilla/Template/Directive.pm

```diff
@@ -0,0 +1,70 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+# This exists to implement the template-before_process hook.
+package Bugzilla::Template::Directive;
+
+use 5.10.1;
+use strict;
+use warnings;
+
+use base qw(Template::Directive);
+
+sub filter {
+    package Template::Directive;
```

Comment:
I also talked to some people at TPC, and while this is a bit crazy it's not that unusual.